WordPress/src/main/java/org/wordpress/android/ui/compose/components/menu/dropdown/Item.kt

```diff
@@ -0,0 +1,100 @@
+package org.wordpress.android.ui.compose.components.menu.dropdown
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.MenuDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.wordpress.android.R
+import org.wordpress.android.ui.compose.theme.AppColor
+import org.wordpress.android.ui.compose.theme.AppTheme
+
+@Composable
+fun Item(
```

Comment:
Initially it was called that (with the prefix) and then I ended up removing it when I created a specific package for the component (`org.wordpress.android.ui.compose.components.menu.dropdown`). I did it to follow the same standard used by the library we're using (Cascade) and only added the prefix to the actual drop-down menu (`JetpackDropdownMenu`).

e.g.
`me.saket.cascade.CascadeColumnScope#DropdownMenuItem` and `androidx.compose.material3.DropdownMenuItem`

When we have a specific package I like the idea of not adding prefix to everything, since it makes things clearer, like:
`Item`
`SubMenu`
`SubMenuHeader`
etc.
VS.
`JetpackDropdownMenuItem`
`JetpackDropdownMenuSubMenu`
`JetpackDropdownMenuSubMenuHeader`

Also I don't think it helps having just one or two items with the prefix, so IMHO we should go with either everything or nothing using a prefix.

Reply:
> or changing this composable visibility to private and moving it inside JetpackDropdownMenu

Also, if we do this to the item shouldn't we also do it with `SubMenu`, `SubMenuHeader`, etc. since only `JetpackDropdownMenu` is going to use it? I'm a bit concerned about the file size.

Reply:
I think those are valid points but I personally think that Cascade's naming is a big downside. For instance, I needed to look into the imports for a couple of files in this PR to understand why they were using a component with the same name (`DropdownMenuItem`) but different fields, so I don't know if that lib is where we should draw inspiration from in that regard.

Honestly, this problem is a bit of Compose's fault since it revolves around functions and not classes and it clearly pushes developers to use top-level Composable functions, so I think it can get pretty messy for general concepts that could have different meaning depending on context (e.g.: `Button`, `Item`, etc) and that's why I prefer the descriptive naming (e.g.: `TextButton`, `OutlineButton`, `MenuItem`) or some kind of strong scope delimitation (e.g.: `private` modifier or "umbrella" `object` around same scope functions like `JetpackDropdownMenu.SubMenu`) and not just package.

I mean, even in our codebase we have problems with top-level Composables with the same name in different packages, such as `PrimaryButton` (we have at least 3) and every once in a while someone uses the wrong `PrimaryButton` because it's easy to miss the package when importing in the IDE.

Maybe we should get a third opinion on this? It could be something we do after-merge as well, I don't think the naming decision needs to block this PR.

Reply:
I still personally prefer having a meaningful package where we don't have to add prefixes everywhere, sometimes making a class name huge and less readable, but I understand the argument regarding top-level Composables and also the lack of package private visibility modifier in Kotlin.

> Maybe we should get a third opinion on this? It could be something we do after-merge as well, I don't think the naming decision needs to block this PR.

I think it's OK, we can sort this out. I can add the prefixes but wanted to ask something else: if we add the prefixes, is there still value in moving them all to the same file (`JetpackDropdownMenu`)? Also, if I move everything to the same file and make the Composables private, do we still need to add the prefix?

Reply:
> My point is: is there value in doing both? Shouldn't we either add prefixes to make them clearer (and less likely for someone to use it by accident) or keep the short names and make them private within the same file?

I don't think there's value in doing both, it's either-or in this case for me. If the Composables are file-private then they are only used in that file and context, so it's pretty clear what's going on, so I completely agree that the names should be short! Otherwise, if they will be on their own separate file then I think it makes sense for the function name to be more descriptive and longer, since they will need to be public.

> also the lack of package private visibility modifier in Kotlin.
Yep, a package-private visibility would get rid of this issue! 😞

Reply:
As discussed in DM, I've pushed a commit moving the menu elements composables to the `JetpackDropdownMenu` file and made them private.